src/main/java/uk/ac/ebi/biostudies/config/IndexManager.java

```diff
@@ -62,13 +62,16 @@ public class IndexManager implements DisposableBean {
     IndexTransferer indexTransferer;
     private IndexReader indexReader;
     private IndexReader fileIndexReader;
+    private IndexReader linkIndexReader;
     private IndexSearcher indexSearcher;
     private IndexSearcher fileIndexSearcher;
+    private IndexSearcher linkIndexSearcher;
     private IndexWriter indexWriter;
     private IndexWriter fileIndexWriter;
-    private Directory indexDirectory, fileIndexDirectory;
-    private IndexWriterConfig indexWriterConfig, fileIndexWriterConfig;
-    private SnapshotDeletionPolicy mainIndexSnapShot, fileIndexSnapShot;
+    private IndexWriter linkIndexWriter;
```

Comment:
Let's globally replace for anything related to extracted links by explictly calling then extracted. So
`linkIndexReader`=>`extractedLinkIndexReader` and so on